exercises/sgf-parsing/canonical-data.json

```diff
@@ -170,6 +170,203 @@
+                    "B": ["bar"]
+                },
+                "children": [
+                    {
+                        "properties": {
+                            "C": ["baz"]
+                        },
+                        "children": []
+                    }
+                ]
+            }
+        },
+        {
+            "uuid": "0b57a79e-8d89-49e5-82b6-2eaaa6b88ed7",
+            "description": "parentheses in property value don't need to be escaped",
+            "property": "parse",
+            "input": {
+                "encoded": "(;A[x(y)z][foo]B[bar];C[baz])"
+            },
+            "expected": {
+                "properties": {
+                    "A": ["x(y)z", "foo"],
+                    "B": ["bar"]
+                },
+                "children": [
+                    {
+                        "properties": {
+                            "C": ["baz"]
+                        },
+                        "children": []
+                    }
+                ]
+            }
+        },
+        {
+            "uuid": "c72a33af-9e04-4cc5-9890-1b92262813ac",
+            "description": "escaped tab in property value is converted to space",
+            "property": "parse",
+            "input": {
+                "encoded": "(;A[hello\\\tworld])"
+            },
+            "expected": {
+                "properties": {
+                    "A": ["hello world"]
+                },
+                "children": []
+            }
+        },
+        {
+            "uuid": "3a1023d2-7484-4498-8d73-3666bb386e81",
+            "description": "escaped newline in property value is converted to nothing at all",
+            "property": "parse",
+            "input": {
+                "encoded": "(;A[hello\\\nworld])"
+            },
+            "expected": {
+                "properties": {
+                    "A": ["helloworld"]
+                },
+                "children": []
+            }
+        },
+        {
+            "uuid": "25abf1a4-5205-46f1-8c72-53273b94d009",
+            "description": "escaped t and n in property value are just letters, not whitespace",
+            "property": "parse",
+            "input": {
+                "encoded": "(;A[\\t = t and \\n = n])"
+            },
+            "expected": {
+                "properties": {
+                    "A": ["t = t and n = n"]
+                },
+                "children": []
+            }
+        },
+        {
+            "comments": [
+                "Keep in mind the difference between:",
+                "1. the string as it is represented in a string literal",
+                "2. the string as it will be presented to the SGF parser",
+                "In particular, the SGF parser will see a property (between the square brackets) with these characters:",
+                "Escaped closing bracket (a single backslash followed by a closing bracket): Insert closing bracket in property value",
```

Comment:
this is verbose, but I am thinking it may be a good idea to avoid backslashes in the comments field if it will prevent further confusion.

Reply:
Absolutely.